weld/subsystem/src/main/java/org/jboss/as/weld/injection/InjectionTargets.java

```diff
@@ -65,7 +69,13 @@ private InjectionTargets() {
              * If this happens, use a combination of a bean archive identifier and class' classloader hashCode as the BDA ID.
              * This breaks AnnotatedType serialization but that does not matter as these are non-contextual components.
              */
-            final String bdaId = beanManager.getId() + componentClass.getClassLoader().hashCode();
+            final ClassLoader classLoader;
+            if (WildFlySecurityManager.isChecking()) {
+                classLoader = AccessController.doPrivileged((PrivilegedAction<ClassLoader>) () -> componentClass.getClassLoader());
```

Comment:
@soul2zimate We should not use lamdas generally. Also there is a GetClassLoaderAction which can be used to do the work for the privileged case

Reply:
I have updated this to use GetClassLoaderAction.